spec-tests/src/test/java/org/candlepin/spec/SystemAdminSpecTest.java

```diff
@@ -0,0 +1,358 @@
+        assertThat(userClient.owners().listOwnerConsumers(owner.getKey()))
+            .singleElement()
+            .returns(consumer1.getUuid(), x -> x.getUuid());
+        assertNotFound(() -> userClient.consumers().getConsumer(consumer2.getUuid()));
+    }
+
+    @Test
+    public void shouldListPoolsForOnlyTheirSystems() throws Exception {
+        OwnerDTO owner = ownerApi.createOwner(Owners.random());
+        UserDTO user = createUserType1(owner);
+        ApiClient userClient = ApiClients.basic(user);
+
+        ConsumerDTO consumer1 = userClient.consumers().createConsumer(
+            Consumers.random(owner).username(user.getUsername()));
+        ConsumerDTO consumer2 = client.consumers().createConsumer(
+            Consumers.random(owner).username("admin"));
+
+        assertThat(userClient.pools().listPoolsByConsumer(consumer1.getUuid())).hasSize(0);
+        assertNotFound(() -> userClient.pools().listPoolsByConsumer(consumer2.getUuid()));
+    }
+
+    @Test
+    public void shouldUnregisterOnlyTheirSystems() throws Exception {
+        OwnerDTO owner = ownerApi.createOwner(Owners.random());
+        UserDTO user = createUserType1(owner);
+        ApiClient userClient = ApiClients.basic(user);
+
+        ConsumerDTO consumer1 = userClient.consumers().createConsumer(
+            Consumers.random(owner).username(user.getUsername()));
+        ConsumerDTO consumer2 = client.consumers().createConsumer(
+            Consumers.random(owner).username("admin"));
+
+        userClient.consumers().deleteConsumer(consumer1.getUuid());
+        assertNotFound(() -> userClient.consumers().deleteConsumer(consumer2.getUuid()));
+    }
+
+    @Test
+    public void shouldCreateEntitlementsOnlyForTheirSystems() {
+        OwnerDTO owner = ownerApi.createOwner(Owners.random());
+        UserDTO user = createUserType1(owner);
+        ApiClient userClient = ApiClients.basic(user);
+
+        ConsumerDTO consumer1 = userClient.consumers().createConsumer(
+            Consumers.random(owner).username(user.getUsername()));
+        ConsumerDTO consumer2 = client.consumers().createConsumer(
+            Consumers.random(owner).username("admin"));
+
+        PoolDTO pool = createPool(owner);
+        userClient.consumers().bindPool(consumer1.getUuid(), pool.getId(), 1);
+        assertNotFound(() -> userClient.consumers().bindPool(consumer2.getUuid(), pool.getId(), 1));
+    }
+
+    @Test
+    public void shouldCreateEntitlementsAsynchronously() throws Exception {
+        OwnerDTO owner = ownerApi.createOwner(Owners.random());
+        UserDTO user = createUserType1(owner);
+        ApiClient userClient = ApiClients.basic(user);
+
+        ConsumerDTO consumer1 = userClient.consumers().createConsumer(
+            Consumers.random(owner).username(user.getUsername()));
+
+        PoolDTO pool = createPool(owner);
+        AsyncJobStatusDTO job = AsyncJobStatusDTO.fromJson(userClient.consumers().bind(consumer1.getUuid(),
```

Comment:
You can use bindPool to get rid of the nulls. There is no need to deserialize the job. You can wait for it by id. Also we should probably assert that entitlements were created.

Reply:
the bindPool in ConsumerClient is async false. I need it to be true.